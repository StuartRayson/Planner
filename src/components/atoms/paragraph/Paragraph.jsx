import classNames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

const Paragraph = ({ className, children }) => {
  const classes = classNames(
    'a-paragraph',
    className
  );

  return (
    <p className={classes}>
      {children}
    </p>
  );
};

Paragraph.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Paragraph;
