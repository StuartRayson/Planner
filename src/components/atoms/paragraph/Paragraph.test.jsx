import React from 'react';
import { shallow } from 'enzyme';

import Paragraph from './Paragraph.jsx';

const caption = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';

describe('<Paragraph />', () => {
  it('renders without crashing', () => {
    shallow(<Paragraph>{caption}</Paragraph>);
  });

  it('should render with correct caption', () => {
    const paragraph = shallow(<Paragraph className="a-paragraph--large">{caption}</Paragraph>);
    expect(paragraph.text()).toEqual(caption);
  });

  it('should render and match snapshot with basic config', () => {
    const wrapper = shallow(
      <Paragraph>{caption}</Paragraph>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
