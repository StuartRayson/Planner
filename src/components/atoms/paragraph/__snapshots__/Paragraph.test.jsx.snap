// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> should render and match snapshot with basic config 1`] = `
<p
  className="a-paragraph"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</p>
`;
